java/sample-container/src/main/java/org/apache/shindig/wso2/ext/impl/WhitelistImpl.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.shindig.wso2.ext.impl;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.shindig.common.uri.Uri;
+import org.apache.shindig.gadgets.http.HttpRequest;
+import org.apache.shindig.wso2.ext.Whitelist;
+import org.wso2.carbon.base.ServerConfiguration;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Very simple implementation of whitelist.
+ * Iterate the URL over a list and checks if it matches.
+ *
+ */
+public class WhitelistImpl implements Whitelist {
+
+    private static final Logger log = Logger.getLogger(WhitelistImpl.class.getName());
+    private List<String> whiteListedBackendURIs = new ArrayList<String>();
+
+    public WhitelistImpl() {
+
+        ServerConfiguration serverConfig = CarbonUtils.getServerConfiguration();
+        String carbonHostname = serverConfig.getFirstProperty("HostName");
+
+        String url = "https://" + carbonHostname + "/portal";
+        addURIRegexToWhiteList(url);
+        log.info("URL: " + url + " added to the whitelisted backend URI list.");
+    }
+
+    public boolean isWhitelisted(HttpRequest request) {
+
+        if (whiteListedBackendURIs.isEmpty()) {
+            return true;
```

Comment:
shouldn't this return false?
If there are no whitelisted URIs then all URIs should be denied, isn't it?

Reply:
Mm... Then we won't be able to allow all URLs if we needed to do so right? Anyway, I think this will not get hit as we're feeding the carbon URL in the constructor.

Reply:
But this goes against the definition of whitelisting.

Reply:
In that case white listed URLs should denote it allows all, something like '*'